subsys/bluetooth/host/mesh/shell.c

```diff
@@ -354,47 +384,71 @@ static const struct bt_mesh_prov prov = {
 	.input = input,
 };
 
-static int cmd_reset(int argc, char *argv[])
+static int cmd_static_oob(int argc, char *argv[])
 {
-	bt_mesh_reset();
-	printk("Local node reset complete\n");
+	if (argc < 2) {
+		prov.static_val = NULL;
+		prov.static_val_len = 0;
+	} else {
+		prov.static_val_len = hex2bin(argv[1], static_val, 16);
+		if (prov.static_val_len) {
+			prov.static_val = static_val;
+		} else {
+			prov.static_val = NULL;
+		}
+	}
+
+	if (prov.static_val) {
+		printk("Static OOB value set (length %u)\n",
+		       prov.static_val_len);
+	} else {
+		printk("Static OOB value cleared\n");
+	}
+
 	return 0;
 }
 
-static bool str2bool(const char *str)
+static int cmd_uuid(int argc, char *argv[])
 {
-	return (!strcmp(str, "on") || !strcmp(str, "enable"));
-}
+	u8_t uuid[16];
+	size_t len;
 
-static u8_t hex2val(char c)
-{
-	if (c >= '0' && c <= '9') {
-		return c - '0';
-	} else if (c >= 'a' && c <= 'f') {
-		return c - 'a' + 10;
-	} else if (c >= 'A' && c <= 'F') {
-		return c - 'A' + 10;
-	} else {
-		return 0;
+	if (argc < 2) {
+		return -EINVAL;
 	}
-}
 
-static size_t hex2bin(const char *hex, u8_t *bin, size_t bin_len)
-{
-	size_t len = 0;
+	len = hex2bin(argv[1], uuid, sizeof(uuid));
+	if (len < 1) {
+		return -EINVAL;
+	}
 
-	while (*hex && len < bin_len) {
-		bin[len] = hex2val(*hex++) << 4;
+	memcpy(dev_uuid, uuid, len);
+	memset(dev_uuid + len, 0, sizeof(dev_uuid) - len);
 
-		if (!*hex) {
-			len++;
-			break;
-		}
+	printk("Device UUID set\n");
 
-		bin[len++] |= hex2val(*hex++);
+	return 0;
+}
+
+static int cmd_reset(int argc, char *argv[])
+{
+	bt_mesh_reset();
+	printk("Local node reset complete\n");
+	return 0;
+}
+
+static u8_t str2u8(const char *str)
```

Comment:
Same for those, could be useful to other shell users

Reply:
I think this one is a bit too mesh specific, since it caters for the 8-bit configuration model states.